### PR TITLE
test: add tests

### PR DESCRIPTION
**Pull request by bvestli**
This will add unit and integration tests using terratest and GitHub workflows to run them automatically

### CODE REVIEW

**Comment by tommybo-entur** [comment]
Jeg tenker at følgende (defaults) bør verifiseres under test:

- force destroy er false
- storage purpose er standard
- label for disable_offsite_backup ikke er satt
- objektversjonering er aktivert
- retention på utdaterte objekter er 2 dager
- config map opprettes

**Comment by bvestli** [comment]
Unit test updated. Ready for review.

File: tests/cloud_storage_unit_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	// Location
+	assert.Equal(t, cloudStorageChanges.AttributeValues["location"], "EUROPE-WEST1", "Incorrect location")
+	// force destroy defalts to false
+	assert.Equal(t, cloudStorageChanges.AttributeValues["force_destroy"], false, "Force destroy should be false")
+	// storage class is standard
+	assert.Equal(t, cloudStorageChanges.AttributeValues["storage_class"], "STANDARD", "Storage class should be type STANDARD")
+	// label for disable_offsite_backup should not exist
+	labels := cloudStorageChanges.AttributeValues["labels"]
+	disable_offsite_backup := labels.(map[string]interface{})["disable_offsite_backup"]
+	assert.Emptyf(t, disable_offsite_backup, "Fount label disable_offsite_backup. This should only exist when set in a production environment.")
```
Comment (tommybo-entur) [inline tests/cloud_storage_unit_test.go:43]: Fount > Found

**Comment by tommybo-entur** [comment]
Bør kanskje skrive en setning om hvordan man kjører enhetstesten lokalt?

**Review by tommybo-entur — CHANGES_REQUESTED** [review]

File: tests/cloud_storage_unit_test.go (hunk 1 above)
Comment (tommybo-entur) [reply tests/cloud_storage_unit_test.go:43]: Kommentaren om hvordan man kjører enhetstesten lokalt er også en del av request for change ;-)

File: .github/workflows/pr-int-test-terraform.yml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 name: Terraform Integration Tests
 
 on:
+  push:
```
Comment (tommybo-entur) [inline .github/workflows/pr-int-test-terraform.yml:4]: Denne gjør at testen ikke blir kjørt ved endringer kun i testen.

**Review by tommybo-entur — APPROVED** [review]
Very good.
A couple of comments, though (for reflection):
I do not see the gain of limiting when to run the build. It introduce corner cases of pushing code that may change behavior without running the tests.
I do prefer testing requirements separately, that is asserting one requirement per test. For regressions it's delightful when the test pinpoint broken requirement and additionally by having all asserts in one test you risk hiding errors as the test ends on first failure.
